.github/workflows/test_macos.yml

```diff
@@ -25,10 +25,10 @@ jobs:
       with:
         fetch-depth: 0
 
-    - name: Set up Python 3.7
+    - name: Set up Python 3.8
       uses: actions/setup-python@v5
       with:
-        python-version: 3.7
+        python-version: 3.8
```

Comment:
I submitted separate PR for this (#7586) to expedite but I believe here we need at least 3.10

Reply:
I'll go for 3.11, and rebase this branch in a few mins.